web-marketplace/src/pages/CreateOrganization/hooks/useCreateDao/useCreateDao.utils.ts

```diff
@@ -0,0 +1,628 @@
+) => [
+  {
+    name: 'Owner',
+    metadata: 'Owner of the project',
+    authorizations: [
+      ...ownerMembersAuthorizations(cw4GroupAddress, rbamAddress),
+      ...projectAdminAuthorizations,
+    ],
+    assignments: [initialOwnerAddress],
+  },
+  {
+    name: 'Admin',
+    metadata:
+      'Manages user access and can edit all project info and project credits.',
+    authorizations: [
+      ...adminMembersAuthorizations(
+        initialOwnerAddress,
+        cw4GroupAddress,
+        rbamAddress,
+      ),
+      ...projectAdminAuthorizations,
+    ],
+  },
+  {
+    name: 'Editor',
+    metadata:
+      'Can edit all project page info and posts. Cannot manage users or credits.',
+    authorizations: [projectsAuthorization, dataAuthorization],
+  },
+  {
+    name: 'Author',
+    metadata:
+      'Can create, edit, and delete their own data posts. Cannot see private post data.',
+    authorizations: [dataAuthorization],
+  },
+  {
+    name: 'Viewer',
+    metadata:
+      'Viewer of the organization, can view all data across all projects, even when private.',
+  },
+];
+
+/**
+ * Parameters required to predict the address of a contract instantiation.
+ */
+type PredictAddressParams = {
+  /**
+   * A client capable of retrieving code details for a code ID.
+   */
+  client: WasmCodeClient;
+  /**
+   * The code ID of the contract to be instantiated.
+   */
+  codeId: number;
+  /**
+   * The address of the contract creator.
+   */
+  creator: string;
+  /**
+   * React Query client used to cache checksum lookups.
+   */
+  queryClient: QueryClient;
+  /**
+   * Optional RPC endpoint identifier to scope the cache key.
+   */
+  rpcEndpoint?: string | null;
+};
+
+export const predictAddress = async ({
+  client,
+  codeId,
+  creator,
+  queryClient,
+  rpcEndpoint,
+}: PredictAddressParams) => {
+  const salt = toUtf8(nanoid());
+  const checksum = await resolveCodeChecksum(
+    client,
+    codeId,
+    queryClient,
+    rpcEndpoint,
+  );
+  const predictedAddress = instantiate2Address(
+    fromHex(checksum),
+    creator,
+    salt,
+    chainInfo.bech32Config.bech32PrefixAccAddr,
+  );
+
+  return { salt: toBase64(salt), predictedAddress };
+};
+
+// from dao-dao-ui
+
+export const encodeJsonToBase64 = (object: any) =>
+  toBase64(toUtf8(JSON.stringify(object)));
+
+async function resolveCodeChecksum(
+  client: WasmCodeClient,
+  codeId: number,
+  queryClient: QueryClient,
+  rpcEndpoint?: string | null,
+): Promise<string> {
+  const staticChecksum = lookupContractChecksum(codeId);
+  if (staticChecksum) {
+    return staticChecksum;
+  }
+
+  const queryOptions = getCodeDetailsQuery({ client, codeId, rpcEndpoint });
+
+  let codeDetails: CodeDetailsLike | undefined;
+  const anyClient = queryClient as any;
+  if (typeof anyClient.ensureQueryData === 'function') {
+    codeDetails = await anyClient.ensureQueryData(queryOptions);
+  } else {
+    codeDetails = await queryClient.fetchQuery(queryOptions);
+  }
```

Comment:
why do we need this logic?
I think you could use our existing `getFromCacheOrFetch` instead